### PR TITLE
Fix #142: introduced pre commit linter

### LINKED ISSUE #142: Introduce pre commit linter
After adding of eslint and fixing of all the files, we need to have eslint to check on pre-commit.

### PR DESCRIPTION
**Pull request by nlok5923**
Fixes: #142 

## Explanation

This pr make use of eslint as a precommit linter

## Checklist
- [ ] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

**Comment by nlok5923** [comment]
@jameesjohn sir please do review this pr and let me know iif some more changes needed.

File: package.json [hunk 1]
```diff
@@ -60,7 +60,8 @@
   "lint-staged": {
     "actions/**/*.+(js|json)": [
       "npm run update-actions-build"
-    ]
+    ],
+    "*.+(js/json) ./spec/**/*.+(js/json) ./lib/*.js ./fixtures/*.js ./actions_build/*.js actions/**/*.+(js|json) ": "npm run lint"
```
Comment (jameesjohn) [inline package.json:64]: We do not need to lint `actions_build/` since that folder is auto-generated.

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks, @nlok5923!
Just one comment, PTAL.

**Comment by nlok5923** [comment]
@jameesjohn i had addressed the request changes please review.

File: package.json (hunk 1 above)
Comment (nlok5923) [reply package.json]: removed

**Comment by jameesjohn** [comment]
For some reason, I'm not able to get the linter to run locally following your command.
Have you tested this locally?

Also, update your branch to fix the tests.

**Comment by nlok5923** [comment]
@jameesjohn please review  had tested lint is working now and branch has been updated.

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "repository": "https://github.com/oppiabot/oppiabot.git",
   "scripts": {
     "dev": "nodemon",
-    "lint": "eslint",
+    "lint": "eslint *.+(js/json) ./spec/**/*.json ./lib/*.js ./fixtures/*.json actions/**/*.js --fix",
```
[the remark this answers is not in the document]
Comment (nlok5923) [reply package.json:10]: Okay sir will correct it in the latest commit.

[the remark this answers is not in the document]
Comment (nlok5923) [reply package.json]: sir please review  again that error was due to a typo.

File: package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
   "repository": "https://github.com/oppiabot/oppiabot.git",
   "scripts": {
     "dev": "nodemon",
-    "lint": "eslint",
+    "lint": "eslint *.js *.json ./spec/**/*.json ./lib/*.js ./fixtures/*.json actions/**/*.js --fix",
```
Comment (jameesjohn) [inline package.json:10]: I think we should exclude JSON files.

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks, final comments

File: package.json (hunk 3 above)
Comment (nlok5923) [reply package.json:10]: Sure sir.

Comment (nlok5923) [reply package.json]: changes addressed in latest commit sir please review.

**Comment by nlok5923** [comment]
@jameesjohn sir the final changes has been addressed please review the pr.

**Review by jameesjohn — APPROVED** [review]
LGTM

**Comment by jameesjohn** [comment]
Can you push the fix for the failing lint check in this PR?
So that users would be able to commit changes after this gets merged.

**Comment by nlok5923** [comment]
Sure sir

**Comment by nlok5923** [comment]
@jameesjohn sir please review the changes

**Comment by jameesjohn** [comment]
There are still some lint errors

**Comment by nlok5923** [comment]
extremely sorry sir correcting those

**Comment by nlok5923** [comment]
@jameesjohn finally i had fixed all the lint issue.

**Comment by jameesjohn** [comment]
I'm investigating the failing tests.

**Comment by jameesjohn** [comment]
Hi, can you merge from master?

**Comment by nlok5923** [comment]
Sure

**Comment by nlok5923** [comment]
> Hi, can you merge from master?

done sir

File: spec/checkPullRequestBranchSpec.js [hunk 4]
```diff
@@ -153,11 +153,11 @@ describe('Pull Request Branch Check', () => {
           'Hi @' +
           author +
           ', PRs made from a branch whose name is prefixed with develop ' +
-          'are not allowed. Also PRs made from develop branch or from a branch '+
-          'whose name is prefixed with develop, release or test are not '+
-          'allowed. So this PR is being closed. Please make your changes ' +
-          'in another branch and send in the PR. To learn more about ' +
-          'contributing to Oppia, take a look at our ' +
+          'are not allowed. Also PRs made from develop branch or from a ' +
+          'branch  whose name is prefixed with develop, release or test ' +
```
Comment (jameesjohn) [inline spec/checkPullRequestBranchSpec.js:157]:
```suggestion
          'branch whose name is prefixed with develop, release or test ' +
```

File: spec/checkPullRequestBranchSpec.js [hunk 5]
```diff
@@ -202,11 +202,11 @@ describe('Pull Request Branch Check', () => {
           'Hi @' +
           author +
           ', PRs made from a branch whose name is prefixed with release ' +
-          'are not allowed. Also PRs made from develop branch or from a branch '+
-          'whose name is prefixed with develop, release or test are not '+
-          'allowed. So this PR is being closed. Please make your changes ' +
-          'in another branch and send in the PR. To learn more about ' +
-          'contributing to Oppia, take a look at our ' +
+          'are not allowed. Also PRs made from develop branch or from a ' +
+          'branch  whose name is prefixed with develop, release or test ' +
```
Comment (jameesjohn) [inline spec/checkPullRequestBranchSpec.js:206]:
```suggestion
          'branch whose name is prefixed with develop, release or test ' +
```

**Review by jameesjohn — CHANGES_REQUESTED** [review]

**Review by jameesjohn — APPROVED** [review]
LGTM

**Pull request merged**